File: bindings/java-ffm/src/main/java/org/maplibre/nativeffi/render/MetalContextDescriptor.java
package org.maplibre.nativeffi.render;

import java.util.Objects;

/** Mutable Metal backend context descriptor. */
public final class MetalContextDescriptor {
  private NativePointer device = NativePointer.NULL;

  public MetalContextDescriptor() {}

  public MetalContextDescriptor(NativePointer device) {
    this.device = Objects.requireNonNull(device, "device");
  }

  public NativePointer device() {
    return device;
  }

  public MetalContextDescriptor device(NativePointer device) {
    this.device = Objects.requireNonNull(device, "device");
    return this;
  }
}

File: bindings/kotlin/src/commonMain/kotlin/org/maplibre/nativeffi/render/MetalContextDescriptor.kt
package org.maplibre.nativeffi.render

/** Mutable Metal backend context descriptor. */
public class MetalContextDescriptor(device: NativePointer) {
  /**
   * Borrowed `MTLDevice`.
   *
   * Texture sessions require a non-null device. Surface attach accepts a null device and then uses
   * `MTLCreateSystemDefaultDevice()`. Surface set-target ignores a null device and keeps the
   * attached device.
   */
  public var device: NativePointer = device
}
